scripts/post-release-notes-preview.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/local/bin/python3
+
+import os
+import requests
+import subprocess
+import sys
+
+BRANCH_PREFIXES_TO_POST = ['feat', 'fix', 'chore', 'test', 'docs', 'revert']
+GIT_ORG = os.environ['CIRCLE_PROJECT_USERNAME']
+GIT_REPO = os.environ['CIRCLE_PROJECT_REPONAME']
+GIT_BRANCH = os.environ['CIRCLE_BRANCH']
+gitBranchParts = GIT_BRANCH.split('/')
+if len(gitBranchParts) <= 1:
+    sys.exit(0)
+
+gitBranchPrefix = gitBranchParts[0]
+if not gitBranchPrefix in BRANCH_PREFIXES_TO_POST:
+    sys.exit(0)
+
+def main():
+    # last version:
+    lastVersion = subprocess.check_output(['git', 'describe', '--abbrev=0', '--tags', 'origin/staging'], text=True).strip()
+
+    # commits since last version
+    commitsSinceLastVersion = subprocess.check_output(['git', 'log', '--no-decorate', lastVersion + '..HEAD', '--oneline'], text=True).strip()
+    commits = processCommits(commitsSinceLastVersion)
+
+    if not (commits['features'] or commits['fixes']):
+        sys.exit(0)
+
+    pullRequestUrls = os.environ.get('CIRCLE_PULL_REQUESTS', '')
+    if not pullRequestUrls:
+        sys.exit(0)
+
+    # TODO: support multiple PRs
```

Comment:
What does this mean? When can it occur?

Reply:
- push commit to branch b1
- open PR from b1 to b2
- open PR from b1 to b3
🎉